internal/configsource/includeconfigsource/README.md

```diff
@@ -1,19 +1,65 @@
 # Include Config Source (Alpha)
 
 Use the include config source to inject [golang templates](https://pkg.go.dev/text/template)
-into the configuration.
+or plain files into the configuration.
 
 ## Configuration
 
 Under the `config_sources:` use `include:` or `include/<name>:` to create a
 template config source.
+The following parameters are available to customize the include config sources:
 
 ```yaml
 config_sources:
   include:
+  include/my_name_00:
+    # delete_files can be used to make the "include" config source to delete the
+    # files referenced by it. This is typically used to remove secrets from the
+    # file system. The file is deleted as soon as its value is read by the config
+    # source. The default value is false. It is an invalid configuration to set it
+    # to true together with the watch_files parameter (see below).
+    delete_files: true
+  include/my_name_01:
+    # watch_files can be used to make the "include" config source monitor for
+    # updates on the used files. Setting it to true will trigger a configuration
+    # reload if any of the files used by the config source are updated.
+    # Configuration reload causes temporary interruption of the data flow during
+    # the time taken to shut down the current pipeline configuration and start the
+    # new one. The default value is false. It is an invalid configuration to set it
+    # to true together with the delete_files parameter (see above).
+    watch_files: true
 ```
 
-The parameters of a include config source are passed to template to be processed.
+Example on how to use the `delete_files` and `watch_files`:
+```yaml
+config_sources:
+  include/default:
+  include/secret:
+    delete_files: true
+  include/watch_for_updates:
+    watch_files: true
+
+components:
+  component_0:
+    # Default usage: configuration won't be reloaded if the file
+    # '/etc/configs/component_field' is changed.
+    component_field: ${include/default:/etc/configs/component_field} 
+
+  component_1:
+    # 'include/secret' was created with 'delete_files' set to true the
+    # file '/etc/configs/secret' after its value is read. If the deletion
+    # fails the collector won't start.
+    component_field: ${include/secret:/etc/configs/secret} 
```

Comment:
Is the lack of space between the `include` and the filename meaningful?  I'm not sure the difference in usage here and in the `|` multiline blocks w/ templating is clear.

Reply:
The problem is that as soon as you add a space after `:` the YAML sees that as a tag so we can't use spaces after it. On the multiline is different since we treat the subsequent line as YAML with the parameters.